### PR TITLE
Bumping up Scala version to 2.10.5/2.11.6. Fixes #1666, Ref #1980

### LINKED ISSUE #1666: [0.13.7] Natural whitespace parsing breaks XML
Found using community-build - https://jenkins-dbuild.typesafe.com:8499/job/Community-2.11.x-manual/56/console
## steps

```
[scala-stm] [error] name := "scala-stm"
[scala-stm] [error] 
[scala-stm] [error] organization := "org.scala-stm"
[scala-stm] [error] 
[scala-stm] [error] version := "0.7"
[scala-stm] [error] 
[scala-stm] [error] scalaVersion := "2.11.0"
[scala-stm] [error] 
[scala-stm] [error] //scalaBinaryVersion := "2.11.0-M8"
[scala-stm] [error] 
[scala-stm] [error] //crossScalaVersions := Seq("2.10.1-RC2", "2.10.0", "2.9.3", "2.9.2")
[scala-stm] [error] 
[scala-stm] [error] //libraryDependencies += ("org.scalatest" % "scalatest_2.10.0-RC5" % "[1.5,)" % "test")
[scala-stm] [error] libraryDependencies += ("org.scalatest" %% "scalatest" % "[1.5,)" % "test")
[scala-stm] [error] 
[scala-stm] [error] libraryDependencies += ("junit" % "junit" % "4.5" % "test")
[scala-stm] [error] 
[scala-stm] [error] // skip exhaustive tests
[scala-stm] [error] testOptions += Tests.Argument("-l", "slow")
[scala-stm] [error] 
[scala-stm] [error] // test of TxnExecutor.transformDefault must be run by itself
[scala-stm] [error] parallelExecution in Test := false
[scala-stm] [error] 
[scala-stm] [error] // make sure Java classes use 1.6 file format
[scala-stm] [error] javacOptions in (Compile, compile) ++= Seq("-source", "1.6", "-target", "1.6")
[scala-stm] [error] 
[scala-stm] [error] ////////////////////
[scala-stm] [error] // publishing
[scala-stm] [error] 
[scala-stm] [error] pomExtra :=
[scala-stm] [error]   <url>http://nbronson.github.com/scala-stm/</url>
[scala-stm] [error]   <licenses>
[scala-stm] [error]     <license>
[scala-stm] [error]       <name>BSD</name>
[scala-stm] [error]       <url>https://github.com/nbronson/scala-stm/blob/master/LICENSE.txt</url>
[scala-stm] [error]       <distribution>repo</distribution>
[scala-stm] [error]     </license>
[scala-stm] [error]   </licenses>
[scala-stm] [error]   <scm>
[scala-stm] [error]     <connection>scm:git:git@github.com:nbronson/scala-stm.git</connection>
[scala-stm] [error]     <url>git@github.com:nbronson/scala-stm.git</url>
[scala-stm] [error]   </scm>
[scala-stm] [error]   <developers>
[scala-stm] [error]     <developer>
[scala-stm] [error]       <id>nbronson</id>
[scala-stm] [error]       <name>Nathan Bronson</name>
[scala-stm] [error]       <email>ngbronson@gmail.com</email>
[scala-stm] [error]     </developer>
[scala-stm] [error]   </developers>
[scala-stm] [error] 
[scala-stm] [error] publishMavenStyle := true
[scala-stm] [error] 
[scala-stm] [error] publishTo <<= (version) { v: String =>
[scala-stm] [error]     val base = "https://oss.sonatype.org/"
[scala-stm] [error]     if (v.trim.endsWith("SNAPSHOT"))
[scala-stm] [error]       Some("snapshots" at base + "content/repositories/snapshots/")
[scala-stm] [error]     else
[scala-stm] [error]       Some("releases" at base + "service/local/staging/deploy/maven2/")
[scala-stm] [error]   }
[scala-stm] [error] 
[scala-stm] [error] // exclude scalatest from the Maven POM
[scala-stm] [error] pomPostProcess := { xi: scala.xml.Node =>
[scala-stm] [error]     import scala.xml._
[scala-stm] [error]     val badDeps = (xi \\ "dependency") filter {
[scala-stm] [error]       x => (x \ "artifactId").text != "scala-library"
[scala-stm] [error]     } toSet
[scala-stm] [error]     def filt(root: Node): Node = root match {
[scala-stm] [error]       case x: Elem => {
[scala-stm] [error]         val ch = x.child filter { !badDeps(_) } map { filt(_) }
[scala-stm] [error]         Elem(x.prefix, x.label, x.attributes, x.scope, ch: _*)
[scala-stm] [error]       }
[scala-stm] [error]       case x => x
[scala-stm] [error]     }
[scala-stm] [error]     filt(xi)
[scala-stm] [error]   }
[scala-stm] [error] 
[scala-stm] [error] credentials += Credentials(Path.userHome / ".ivy2" / ".credentials")
```
## problem

```
[scala-stm] [error] [/localhome/jenkinsdbuild3/workspace/Community-2.11.x-manual/target-0.9.1/extraction/cdcab9835c609aecf7d943156d53fb28dbd9eb58/projects/e700e61574afb95ee447d356e119d5985ade471b/build.sbt]:52: ';' expected but 'true' found.
[scala-stm] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
[scala-stm:error] java.lang.RuntimeException: 
[scala-stm:error]   at scala.sys.package$.error(package.scala:27)
[scala-stm:error]   at distributed.support.sbt.SbtRunner.distributed$support$sbt$SbtRunner$$processCommand(SbtRunner.scala:93)
[scala-stm:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:74)
[scala-stm:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:74)
[scala-stm:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:74)
[scala-stm:error]   at distributed.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:64)
[scala-stm:error]   at sbt.IO$.withTemporaryFile(IO.scala:311)
[scala-stm:error]   at distributed.support.sbt.SbtRunner.run(SbtRunner.scala:64)
[scala-stm:error]   at distributed.support.sbt.SbtExtractor$.extractMetaData(DependencyExtractor.scala:99)
[scala-stm:error]   at distributed.support.sbt.SbtBuildSystem.extractDependencies(SbtBuildSystem.scala:58)
[scala-stm:error]   at distributed.support.sbt.SbtBuildSystem.extractDependencies(SbtBuildSystem.scala:18)
```
## notes

Scala allows series of XML nodes as NodeSeq.

``` scala
scala> val x = <x/><y/>
x: scala.xml.NodeBuffer = ArrayBuffer(<x/>, <y/>)
```

/cc @ajozwik

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes #1666, Subsumes #1980

To pass `File => Unit` callback across the classloader boundary
I am encoding it as a `java.util.List[File]` by overriding `add(x: File)` method.
This was needed since Java didn't allow me to cast from one classloader to the other.

/review @jsuereth, @dwijnand

### CODE REVIEW

File: scripted/sbt/src/main/scala/sbt/test/ScriptedTests.scala [hunk 1]
```diff
@@ -108,19 +109,27 @@ object ScriptedTests {
...
+  // Using java.util.List[File] to encode File => Unit
   def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], bootProperties: File,
-    launchOpts: Array[String], prescripted: File => Unit): Unit =
-    run(resourceBaseDirectory, bufferLog, tests, ConsoleLogger(), bootProperties, launchOpts, prescripted) //new FullLogger(Logger.xlog2Log(log)))
+    launchOpts: Array[String], prescripted: java.util.List[File]): Unit =
+    run(resourceBaseDirectory, bufferLog, tests, ConsoleLogger(), bootProperties, launchOpts,
+      { f: File => prescripted.add(f); () }) //new FullLogger(Logger.xlog2Log(log)))
 
-  def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], logger: AbstractLogger, bootProperties: File,
+  def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], logger: xsbti.Logger, bootProperties: File,
```
Comment (jsuereth) [inline scripted/sbt/src/main/scala/sbt/test/ScriptedTests.scala]: Does this break the ABI?

Comment (eed3si9n) [inline scripted/sbt/src/main/scala/sbt/test/ScriptedTests.scala]: I can add shims around to keep bincompat, but this module should be used only by sbt itself, so I don't think it matters.

**Comment by jsuereth** [comment]
The change looks good. Just want to make sure the up-scaled interface is ABI compatible.  I think it is.  In practice, given how scripted is used, I think it doesn't matter as well (scripted is usually versioned by the sbt version).

**Comment by eed3si9n** [comment]
Yea. Note I didn't touch the scripted base or scripted plugin. (They may actually encounter similar issue as I have)

**Comment by dwijnand** [comment]
LGTM.

Can the crazy use of java.util.List be removed post 0.13.9 when sbt's build moves to 0.13.9?

**Comment by eed3si9n** [comment]
We need to add `F1` in xsbti now if we want to remove `java.util.List` down the line.

**Pull request merged**